processor/src/main/java/org/mapstruct/ap/internal/model/MappingBuilderContext.java

```diff
@@ -93,7 +93,13 @@
         Assignment getTargetAssignment(Method mappingMethod, Type targetType,
                                        FormattingParameters formattingParameters,
                                        SelectionCriteria criteria, SourceRHS sourceRHS,
-                                       AnnotationMirror positionHint);
+                                       AnnotationMirror positionHint,
+                                       Forger forger);
+
+        @FunctionalInterface
+        interface Forger {
```

Comment:
Why not use `java.util.function.Supplier`? Also if we go with our own, maybe `AssignmentForger`?

Reply:
I went for the `Supplier`.. Perhaps better than inventing our own types..